packages/datasources/src/namespaces.ts

```diff
@@ -59,16 +55,20 @@ export const getDatasource = <
  * or may not actually be defined. For example, if using {@link getDatasource}, with a
  * `namespaceId: ENSNamespaceId`, the typechecker will enforce that the only valid `datasourceName`
  * is ENSRoot (the only Datasource present in all namespaces). This method allows you to receive
- * `Datasource | undefined` for a specified `datasourceName`.
+ * the const Datasource or undefined for a specified `datasourceName`.
  *
- * @param namespaceId - The ENSNamespace identifier (e.g. 'mainnet', 'sepolia', 'holesky', 'ens-test-env')
+ * @param namespaceId - The ENSNamespace identifier (e.g. 'mainnet', 'sepolia', 'ens-test-env')
  * @param datasourceName - The name of the Datasource to retrieve
  * @returns The Datasource object for the given name within the specified namespace, or undefined if it does not exist
  */
-export const maybeGetDatasource = (
-  namespaceId: ENSNamespaceId,
-  datasourceName: DatasourceName,
-): Datasource | undefined => (getENSNamespace(namespaceId) as ENSNamespace)[datasourceName];
+export const maybeGetDatasource = <
```

Comment:
better types, returns the const Datasource you expect or undefined, removes need for "at compile time" version of all of this